extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/kernel/StageDefinition.java

```diff
@@ -130,10 +129,6 @@ public class StageDefinition
     this.shuffleCheckHasMultipleValues = shuffleCheckHasMultipleValues;
     this.frameReader = Suppliers.memoize(() -> FrameReader.create(signature))::get;
 
-    if (mustGatherResultKeyStatistics() && shuffleSpec.clusterBy().getColumns().isEmpty()) {
```

Comment:
```
if (intermediateClusterBy.getColumns().isEmpty() && resultClusterBy.isEmpty()) {
      // Ignore shuffleSpecFactory, since we know only a single partition will come out, and we can save some effort.
      shuffleSpecFactoryPreAggregation = ShuffleSpecFactories.singlePartition();
      shuffleSpecFactoryPostAggregation = ShuffleSpecFactories.singlePartition();
    } else if (doOrderBy) {
      if( intermediateClusterBy.getColumns().isEmpty()){
        shuffleSpecFactoryPreAggregation=ShuffleSpecFactories.singlePartition();
      }else {
        shuffleSpecFactoryPreAggregation = ShuffleSpecFactories.globalSortWithMaxPartitionCount(maxWorkerCount);
      }
      shuffleSpecFactoryPostAggregation = doLimitOrOffset
                                          ? ShuffleSpecFactories.singlePartition()
                                          : resultShuffleSpecFactory;
    } else {
    
    
    
```

I think we should do this instead of removing this check. 
There is no shuffling required in the `GroupByPreShuffleStage`